llama_index/readers/base.py

```diff
@@ -18,3 +22,30 @@ def load_langchain_documents(self, **load_kwargs: Any) -> List[LCDocument]:
         """Load data in LangChain document format."""
         docs = self.load_data(**load_kwargs)
         return [d.to_langchain_format() for d in docs]
+
+
+class PydanticBaseReader(BaseReader, BaseComponent):
+    """Serialiable Data Loader with Pydatnic."""
+
+    is_remote: bool = False
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class LoaderConfig(BaseComponent):
+    """Represents a loader and it's input arguments."""
+
+    loader: BaseReader = Field(..., description="Loader to use.")
```

Comment:
putting notes here from offline discussion:
1. loader ctor (& pydantic attributes) configures "how to connect to API"
2. args/kwargs configures "what data to load"

Reply:
2 corresponds to `def load_data` calls.